src/build/plan.rs

```diff
@@ -60,29 +63,27 @@ crate struct Plan {
     // An object for finding the package which a file belongs to and this inferring
     // a package argument.
     package_map: Option<PackageMap>,
-    // The package argument used in the last Cargo build.
-    prev_package_arg: Option<PackageArg>,
+    /// Packages (names) for which this build plan was prepared.
+    /// Used to detect if the plan can reused when building certain packages.
+    built_packages: HashSet<String>,
```

Comment:
The might be several packages with the same name. Ideally, we should use something like package_id here I think

Reply:
OTOH, I'm all for not peeking into Cargo's internals unless absolutely necessary, and I think using strings for know is the best approach.

Reply:
Agreed on packages with same name; came to the conclusion that having packages with same name in the primary workspace will be rare enough and these strings we're passing to Cargo's `Packages::from_flags`, so I think these get converted and resolved later to `PackageIdSpec`s internally?

Re Cargo internals I'd love to stabilise a small subset of public-facing structures like PackageId but that's a talk for another time 😄